node_app/server.js

```diff
@@ -52,101 +50,71 @@ app.post("/Organization", (req, res) => {
-          title: `Photo of ${patient.given} ${patient.family} ${patient.suffix}`,
-        },
-      ],
-      contact: [
-        {
-          relationship: [
-            {
-              coding: [
-                {
-                  // unsure
-                  system: "http://terminology.hl7.org/CodeSystem/v2-0131",
-                  code: patient.contact.relationship,
-                  display: patient.contact.relationship,
-                },
-              ],
-            },
-          ],
-          name: {
-            family: patient.contact.family,
-            given: [patient.contact.given],
-          },
-          telecom: [
-            {
-              system: "phone",
-              value: patient.contact.phone.value,
-              use: patient.contact.phone.use,
-            },
-          ],
-        },
-      ],
-      communication: [
-        {
-          language: {
-            text: patient.language,
-          },
-          preferred: true,
-        },
-      ],
+  createPatient(req, res).catch((e) =>
+    res.status(400).json({
+      error: e.response ? e.response.data : e.message,
+    })
+  );
+});
+
+async function createPatient(req, res) {
+  let patients = req.body.Patient;
+  let patient = patients.shift();
+  let promises = [];
+  let response = { Patient: [] };
+
+  // resource for head of household
+  if (patient) {
+    let related = { resourceType: "RelatedPerson" };
+    let result = await createPatient_(patient, related);
+    response.Patient.push(result.data);
+
+    // update related
+    let related_id = result.data.link[0].other.reference.split("/")[1];
+    resource = RelatedPerson.toFHIR({
+      patient: `Patient/${result.data.id}`,
+      relationship: patient.relationship,
+    });
+    resource.id = related_id;
+    await axios.put(`${base}/RelatedPerson/${related_id}`, resource, headers);
+  }
+
+  // resources for members
+  for (patient of patients) {
+    let related = {
+      patient: `Patient/${response.Patient[0].id}`,
+      relationship: patient.relationship,
     };
-    // add in telecom
-    for (idx in patient.phone) {
-      resource.telecom.push({
-        system: "phone",
-        value: patient.phone[idx].value,
-        use: patient.phone[idx].use,
-        rank: `${idx}`,
-      });
-    }
-    for (idx in patient.email) {
-      resource.telecom.push({
-        system: "email",
-        value: patient.email[idx],
-        rank: `${idx}`,
-      });
-    }
 
-    // post resource
-    axios
-      .post(`${base}/Patient`, resource, headers)
-      .then((response) => {
-        res.json(response.data);
-      })
-      .catch((e) => res.send(e));
+    promises.push(createPatient_(patient, related));
   }
-});
+
+  Promise.all(promises).then((results) => {
+    for (result of results) {
+      response.Patient.push(result.data);
+    }
+    res.json(response);
+  });
+}
+
+// create patient and link to relatedperson
+async function createPatient_(patient, related) {
```

Comment:
Can we call this "createPatientWithLink" or something? "createPatient" vs "createPatient_" is bound to be confusing when we come back to this later.

Reply:
Done